mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -826,24 +832,36 @@ impl TakerSwap {
         taker_coin_swap_contract: Vec<u8>,
     ) -> NegotiationDataMsg {
         let r = self.r();
-        if r.data.maker_coin_htlc_pubkey != r.data.taker_coin_htlc_pubkey {
-            NegotiationDataMsg::V3(NegotiationDataV3 {
+
+        let maker_coin_has_persistent_pub = r
+            .data
+            .maker_coin_htlc_pubkey
+            .map(|maker_pub| maker_pub == r.data.my_persistent_pub)
+            .unwrap_or(true);
+        let taker_coin_has_persistent_pub = r
+            .data
+            .taker_coin_htlc_pubkey
+            .map(|taker_pub| taker_pub == r.data.my_persistent_pub)
+            .unwrap_or(true);
+
+        if maker_coin_has_persistent_pub && taker_coin_has_persistent_pub {
```

Comment:
This can be simplified to 
```rust
        if r.data.maker_coin_htlc_pubkey == r.data.taker_coin_htlc_pubkey
            && r.data.taker_coin_htlc_pubkey == Some(r.data.my_persistent_pub)
```

Reply:
Done